lib/pascal.rb

```diff
@@ -1,36 +1,33 @@
+class Pascal_triangle
+  def draw(height=1)
+    spaces = " "
+    num_spaces = height
+    for row in (1..height)
+      puts "#{spaces*num_spaces}#{nth_row(row)}"
+      num_spaces -= 1
+    end
+  end
 
-class Pascal
-
-
-  def nth_row(row_num=1)
-    row = [1,1]
-    new_row= []
-
-    if row_num == 1
-      return "1"
-    elsif row_num == 2
-      return row.join(" ")
-    else
-      (row_num-2).times do
-        new_row = []
-        (row.length-1).times do |i|
-            new_row << (row[i]+row[i+1])
-        end
-        row = [1,1]
-        new_row.each do |num|
-          row.insert(1,num)
-        end
-      end
-      return row.join(" ")
+  private
+  def next_row(row)
+    new_values = []
+    (row.length - 1).times do |i|
+      new_values << (row[i] + row[i+1])
+    end
+    new_row = [1,1]
+    new_values.each do |num|
+      new_row.insert(1,num)
     end
+    new_row
   end
 
-  def draw_pascal_triangle(height=1)
-    spaces = " "
-    num_spaces = height
-    for row in(1..height)
-      puts "#{spaces*num_spaces}#{nth_row(row)}"
-      num_spaces -=1
+  def nth_row(row_num = 1)
+    row = [1, 1]
+    return "1" if row_num == 1
+    return row.join(" ") if row_num == 2
+    (row_num-2).times do
```

Comment:
You can use a one-liner syntax for this block `(row_num - 2).times { row = next_row(row) }`. Add white space before and after `-` sign.